internal/provider/resource_jwks_url.go

```diff
@@ -0,0 +1,173 @@
+package provider
+
+import (
+	"context"
+	"errors"
+	"slices"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	neon "github.com/kislerdm/neon-sdk-go"
+)
+
+func resourceJwksUrl() *schema.Resource {
+	return &schema.Resource{
+		Description: `Project JWKS URL. See details: https://neon.tech/docs/guides/neon-rls-authorize
+`,
+		SchemaVersion: 1,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceJwksUrlImport,
+		},
+		CreateContext: resourceJwksUrlCreateRetry,
+		ReadContext:   resourceJwksUrlReadRetry,
+		DeleteContext: resourceJwksUrlDeleteRetry,
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Project ID.",
+			},
+			"jwks_url": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The URL that lists the JWKS.",
+			},
+			"provider_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of the authentication provider.",
+			},
+			"role_names": {
+				Type:        schema.TypeList,
+				MinItems:    1,
+				MaxItems:    10,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The roles the JWKS should be mapped to.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"branch_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Branch ID.",
+			},
+			"jwt_audience": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of the required JWT Audience to be used.",
+			},
+		},
+	}
+}
+
+func updateStateJwksUrl(d *schema.ResourceData, v neon.JWKS, role_names []string) error {
+	if err := d.Set("project_id", v.ProjectID); err != nil {
+		return err
+	}
+	if err := d.Set("jwks_url", v.JwksURL); err != nil {
+		return err
+	}
+	if err := d.Set("provider_name", v.ProviderName); err != nil {
+		return err
+	}
+	// todo - we don't get this back from the API, is using the provided values ok?
```

Comment:
@kislerdm As mentioned in the issue, the [List JWKS URLs endpoint](https://api-docs.neon.tech/reference/getprojectjwks) doesn't return the `role_names`. My thought was that since we have the role names available in the creation configuration, we could still manage that state, but for imported values, we wouldn't be able to support that.

Very open to suggestions, is this the right approach?